src/widgets/SettingsIcon.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2021
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { selectUsingVaccines } from '../selectors/modules';
+import { PermissionSelectors } from '../selectors/permission';
+import { PermissionActions } from '../actions/PermissionActions';
+
+import { useToggle } from '../hooks/index';
+import { CancelIcon, CogIcon, ConfirmIcon, DisabledIcon, HazardIcon } from './icons';
+import { PaperModalContainer } from './PaperModal/PaperModalContainer';
+import { IconButton } from './IconButton';
+import { FlexRow } from './FlexRow';
+
+import { DARK_GREY } from '../globalStyles';
+import { modalStrings } from '../localization';
+
+const PermissionRow = ({ comment, enabled, onPress, label }) => (
+  <TouchableOpacity onPress={enabled ? null : onPress}>
+    <FlexRow alignItems="flex-start" flex={0} style={localStyles.permissionRowContainer}>
+      {enabled ? <ConfirmIcon /> : <DisabledIcon />}
+      <FlexRow style={{ flexDirection: 'column', marginLeft: 10 }}>
+        <Text style={localStyles.permissionRowText}>{label}</Text>
+        <Text style={localStyles.permissionRowComment}>{comment}</Text>
+      </FlexRow>
+    </FlexRow>
+  </TouchableOpacity>
+);
+
+PermissionRow.defaultProps = {
+  comment: '',
+};
+
+PermissionRow.propTypes = {
+  comment: PropTypes.string,
+  enabled: PropTypes.bool.isRequired,
+  label: PropTypes.string.isRequired,
+  onPress: PropTypes.func.isRequired,
+};
+
+const SettingsIconComponent = ({
+  bluetooth,
+  location,
+  requestBluetooth,
+  requestLocation,
+  requestWriteStorage,
+  usingVaccines,
+  writeStorage,
+}) => {
+  const [settingsModalOpen, toggleSettingsModal] = useToggle();
+
+  if (!usingVaccines) {
+    return null;
+  }
+
+  const icon =
+    location && bluetooth && writeStorage ? (
+      <CogIcon color={DARK_GREY} />
+    ) : (
+      <HazardIcon color={DARK_GREY} size={20} />
```

Comment:
😁  I sooo wanted to. I resisted because I didn't want to be too different from the current design.
Your prompting was needed, thanks!
it looks good, I think. I've used `SUSSOL_ORANGE` though, as that matches the `InfoBadge` and menu page icons.

![Screen Shot 2021-02-12 at 9 48 36 AM](https://user-images.githubusercontent.com/9192912/107697539-3e7db300-6d18-11eb-808a-242ddb08a382.png)

I've also demoted the cog from `DARK_GREY` to `GREY` as this is more in keeping with the rest of that header, and also the settings cog down the bottom.

![Screen Shot 2021-02-12 at 9 55 21 AM](https://user-images.githubusercontent.com/9192912/107697717-738a0580-6d18-11eb-9f70-823cc39f26ba.png)

Another concern was whether the modal needed some introductory text, explaining what was happening and that you can press to enable. I think that it does, so have had a go..

![Screen Shot 2021-02-12 at 10 04 00 AM](https://user-images.githubusercontent.com/9192912/107698672-c2846a80-6d19-11eb-8e97-dcd348bd1c85.png)